lib/stencil-bundle.js

```diff
@@ -151,51 +152,98 @@ class Bundle {
         };
     }
 
-    assembleTemplatesTask(callback) {
+    async assembleTemplatesTask(callback) {
         console.log('Template Parsing Started...');
+        const internalTemplatesList = await recursiveReadDir(this.templatesPath, ['!*.html']);
+        const searchExternalLib = (content) => {
```

Comment:
Do you mind moving searchExternalLib and checkTemplate to class functions (and maybe renaming to make more sense in class domain). 
Maybe you can think also about extracting this functionality into separate file, while you are here?

Reply:
done